flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReader.java

```diff
@@ -203,6 +204,10 @@ private boolean isNewlyAddedTableSplitAndBinlogSplit(
 
     @Override
     public void addSplits(List<MySqlSplit> splits) {
+        addSplits(splits, false);
+    }
+
+    private void addSplits(List<MySqlSplit> splits, boolean selfUpdate) {
```

Comment:
it's hard to understand `selfUpdate`. How about 
```suggestion
        /**
     * Adds a list of splits for this reader to read.
     *
     * @param splits the splits to add.
     * @param checkTablesForBinlogSplit to check the captured table list change or not, it should be
     *     true for reader which is during restoration from a checkpoint or savepoint.
     */
    private void addSplits(List<MySqlSplit> splits, boolean checkTableChangeForBinlogSplit) {
    ...
       if (checkTableChangeForBinlogSplit) { 
       filterOutdatedSplitInfos(...)
    
```